### PR TITLE
fix: validate DB-specific parameters

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

The `validate_parameters` endpoint was not performing the DB-specific Marshmallow validation (based on `parameters_schema`), but only the validation from the `validate_parameters` schema in the DB engine spec (which does more deep validation like checking if the host is resolvable and the port is open, but doesn't validate the schema).

I changed the `DatabaseValidateParametersSchema` schema validation to also validate the DB-specific parameters, and changed the API to return the error messages expected by the frontend for the `onBlur` validation (with an `invalid` key in the `extra`).

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

Added unit tests.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/databases/api.py [hunk 1]
```diff
@@ -999,7 +1001,17 @@ def validate_parameters(  # pylint: disable=too-many-return-statements
         try:
             payload = DatabaseValidateParametersSchema().load(request.json)
         except ValidationError as error:
-            raise InvalidPayloadSchemaError(error)
+            # {'port': ['Not a valid integer.']}
```
Comment (hughhhh) [inline superset/databases/api.py:1004]: remove comment

File: tests/databases/api_tests.py [hunk 2]
```diff
@@ -1733,6 +1744,77 @@ def test_validate_parameters_missing_fields(self):
...
+                "query": {},
+            }
+        )
+        rv = self.client.post(url, json=payload)
+        response = json.loads(rv.data.decode("utf-8"))
+
+        assert rv.status_code == 200
+        assert response == {"message": "OK"}
+
+    def test_validate_parameters_invalid_port(self):
```
Comment (hughhhh) [inline tests/databases/api_tests.py:1778]: Can we add validation for the range of ports allowed?

Comment (betodealmeida) [reply tests/databases/api_tests.py:1778]: Sure, Arash is adding that on a separate PR.

File: superset/databases/api.py (hunk 1 above)
Comment (betodealmeida) [reply superset/databases/api.py:1004]:
```suggestion
```

File: tests/databases/api_tests.py (hunk 2 above)
Comment (betodealmeida) [reply tests/databases/api_tests.py:1778]: https://github.com/apache/superset/pull/15151/files#diff-2e62d64ef1113e48efdfeb2acbaa522fca13e49e6a00c2cfd4f74efc4ae1b45cR1314

**Review by eschutho — APPROVED** [review]

**Pull request merged**